sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpError.cs

```diff
@@ -213,16 +212,15 @@ internal static class AmqpError
                 if (NotFoundExpression.IsMatch(description)
                     || (description.IndexOf(NotFoundStatusText, StringComparison.InvariantCultureIgnoreCase) >= 0))
                 {
-                    return new EventHubsResourceNotFoundException(eventHubsResource, description);
+                    return new EventHubsException(eventHubsResource, description, EventHubsException.FailureReason.ResourceNotFound);
                 }
 
-
-                return new EventHubsCommunicationException(eventHubsResource, description);
+                return new EventHubsException(eventHubsResource, description, EventHubsException.FailureReason.ServiceCommunicationProblem);
             }
 
             // There was no specific exception that could be determined; fall back to a generic one.
 
-            return new EventHubsException(true, eventHubsResource, description);
+            return new EventHubsException(eventHubsResource, description, EventHubsException.FailureReason.ServiceCommunicationProblem);
```

Comment:
Since it happened within the scope of an AMQP operation, it should be a communication exception.   Either the service didn't respond properly or the client couldn't get its transport stuff in the right state to communicate.